lib/autoupdate/rollout/transitions.go

```diff
@@ -0,0 +1,251 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package rollout
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	autoupdatev1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1"
+	"github.com/gravitational/teleport/api/types/autoupdate"
+)
+
+const (
+	updateReasonManualTrigger  = "manual_trigger"
+	updateReasonForcedDone     = "manual_forced_done"
+	updateReasonManualRollback = "manual_rollback"
+)
+
+// TriggerGroups mutates a rollout to trigger updates for the given groups.
+// The function does not apply the rollout back, it is the caller's responsibility
+// to commit it in the backend.
+// The function takes a desired State parameter to leave room for future canary
+// state support as specified in RFD 184.
+func TriggerGroups(rollout *autoupdatev1pb.AutoUpdateAgentRollout, groupNames []string, desiredState autoupdatev1pb.AutoUpdateAgentGroupState, now time.Time) error {
+	// Validation part, we look for everything not in order or unsupported.
+	if rollout == nil {
+		return trace.BadParameter("rollout cannot be nil")
+	}
+
+	err := checkSchedule(rollout)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = strategySupportsManualTransition(rollout)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	switch desiredState {
+	case autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_UNSPECIFIED:
+		// When unspecified, we default to active
+		desiredState = autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_ACTIVE
+	case autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_ACTIVE:
+	default:
+		return trace.BadParameter("unsupported desired state: %s, supported states are 'unspecified' and 'active'.", getGroupStateName(desiredState))
```

Comment:
```suggestion
		return trace.BadParameter("unsupported desired state: %s, supported states are 'unspecified' and 'active'", getGroupStateName(desiredState))
```
*nit*: Error should not have terminal punctuation